libs/shared/directives/src/validate-input.directive.ts

```diff
@@ -0,0 +1,58 @@
+import { Directive, ElementRef, inject, OnInit, Renderer2 } from '@angular/core';
+import { AbstractControl, NgControl } from '@angular/forms';
+import { SubscriptionHandler } from '@keira/shared/utils';
+import { ValidationService } from '@keira/shared/common-services';
+
+@Directive({
+  selector: '[keiraInputValidation]',
+  standalone: true,
+})
+export class InputValidationDirective extends SubscriptionHandler implements OnInit {
+  private readonly el: ElementRef = inject(ElementRef);
+  private readonly renderer: Renderer2 = inject(Renderer2);
+  private readonly ngControl: NgControl = inject(NgControl);
+  private readonly validationService = inject(ValidationService);
+
+  private errorDiv: HTMLElement | null = null;
+
+  ngOnInit(): void {
+    const control = this.ngControl.control;
+
+    if (!control) {
+      return;
+    }
+
+    this.subscriptions.push(
+      control.statusChanges?.subscribe(() => {
```

Comment:
reading this, we could put a debounceTime here and/or a distinctUntilChanged in a pipe

Reply:
Done